library/src/main/java/com/pokegoapi/api/settings/Settings.java

```diff
@@ -123,6 +123,23 @@ public void updateSettings(DownloadSettingsResponse response) {
 		if (response.getSettings().hasGpsSettings()) {
 			gpsSettings.update(response.getSettings().getGpsSettings());
 		}
+		// check if new api has been forced by Niantic
+		if(!response.getSettings().getMinimumClientVersion().isEmpty()) {
+			int[] currentVer = api.getHashProvider().getCurrentAPIVersion();
+			String[] curVer = response.getSettings().getMinimumClientVersion().split("\\.");
+			int[] minimumVer = new int[currentVer.length];
+			for(int x = 0; x < currentVer.length; x++)
+				minimumVer[x] = Integer.valueOf(curVer[x]);
```

Comment:
`Integer.valueOf(String)` is for returning an `Integer` object, here you can just use `Integer.parseInt(String)`. You should also handle the `NumberFormatException`.